src/python_testing/TC_CLDIM_3_3.py

```diff
@@ -0,0 +1,535 @@
+            logging.info("Positioning feature or Limitation feature is not supported. Skipping steps 5b to 5g.")
+            self.skip_step("5b")
+            self.skip_step("5c")
+            self.skip_step("5d")
+            self.skip_step("5e")
+            self.skip_step("5f")
+            self.skip_step("5g")
+        else:
+            # STEP 5b: Send SetTarget command with Position 0%
+            self.step("5b")
+            sub_handler.reset()
+            try:
+                await self.send_single_cmd(
+                    cmd=Clusters.Objects.ClosureDimension.Commands.SetTarget(position=0),
+                    endpoint=endpoint
+                )
+            except InteractionModelError as e:
+                asserts.assert_equal(e.status, Status.Success, "Unexpected error returned")
+
+            # STEP 5c: Verify TargetState attribute is updated
+            self.step("5c")
+            target_state = await self.read_cldim_attribute_expect_success(endpoint=endpoint, attribute=attributes.TargetState)
+            asserts.assert_equal(target_state.position, min_position, "TargetState Position does not match MinPosition")
+
+            # STEP 5d: Wait for CurrentState.Position to be updated to 0%
+            self.step("5d")
+            if initial_state.position > 0:
+                sub_handler.await_all_expected_report_matches(
+                    expected_matchers=[current_position_matcher(0)], timeout_sec=timeout)
+            else:
+                logging.info("Initial Position not > 0. Skipping step 5d.")
+                self.mark_current_step_skipped()
+
+            # STEP 5e: Send SetTarget command with Position 100%
+            self.step("5e")
+            sub_handler.reset()
+            if max_position < 10000:
+                try:
+                    await self.send_single_cmd(
+                        cmd=Clusters.Objects.ClosureDimension.Commands.SetTarget(position=10000),
+                        endpoint=endpoint
+                    )
+                except InteractionModelError as e:
+                    asserts.assert_equal(e.status, Status.Success, "Unexpected error returned")
+            else:
+                logging.info("MaxPosition not < 10000. Skipping step 5e.")
+                self.mark_current_step_skipped()
+
+            # STEP 5f: Verify TargetState attribute is updated
+            self.step("5f")
+            if max_position < 10000:
+                target_state = await self.read_cldim_attribute_expect_success(endpoint=endpoint, attribute=attributes.TargetState)
+                asserts.assert_equal(target_state.position, max_position, "TargetState Position does not match MaxPosition")
+            else:
+                logging.info("MaxPosition not < 10000. Skipping step 5f.")
+                self.mark_current_step_skipped()
+
+            # STEP 5g: Wait for CurrentState.Position to be updated to 100%
+            self.step("5g")
+            if max_position < 10000:
```

Comment:
This check is ok to avoid waiting for a no-change. The others ones, i am not convinced.